parquet/sandboxes/heka/output/s3_parquet.lua

```diff
@@ -126,7 +130,8 @@ if type(json_objects) == "table" then
     require "cjson"
     json_objects_len = #json_objects
 end
-if not json_objects and metadata_group then error("metadata_group cannot be configured without json_objects") end
+local heka_msg_use_fields   = read_config("heka_message_use_fields")
+if not json_objects and not heka_msg_use_fields and metadata_group then error("metadata_group cannot be configured without json_objects") end
```

Comment:
This reads a bit weird.. you will need different checks as this one allows both json_objects and heka_msg_use_fields
and a metadata/heka_msg_use_fields combo